docs/DeveloperGuide.md

```diff
@@ -285,134 +285,95 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
-    * 3a3. Student enters new command.
-      Steps 3a1-3a3 are repeated until the command entered are correct.
-      \
-      Use case resumes from step 4.
-
-
-* 4a. MassLinkers detects there is no such tag.
-    * 4a1. MassLinkers prompts error message.
-    * 4a2. MassLinkers requests for the correct command.
-    * 4a3. Student enters new command.
-      \
-      Steps 4a1-4a3 are repeated until the command entered are correct.
-      \
-      Use case resumes from step 4.
-
-
-* a. At any time, Student chooses to close MassLinkers.
-    * a1. MassLinkers updates the written storage file.
-    * a2. MassLinkers closes the program.
+* 2a. The list of tags is empty.
+  * a1. MassLinkers returns an empty list. 
+    \
+  Use case ends.
+* 2b. MassLinkers detects there is no such tag.
+  * b1. MassLinkers requests for another tag.
+  * Step b1 is repeated until a valid tag is found. \
+  Use case resumes from step 2.
+* *a. At any time, Student chooses to close MassLinkers.
+    * a1. MassLinkers updates the storage file.
       \
       Use case ends.
 
 
-**Use case 4: Add a tag to a classmate**
+**Use case 4: Add a tag to a batchmate**
 
 **MSS**
 
-1. Student chooses to add a tag on a classmate, could be himself or others.
-2. Student enters the add tag command with the associated classmate.
-3. MassLinkers verifies the syntax of the command.
-4. MassLinkers searches through the list of people for the selected classmate.
-5. MassLinkers adds tag to the classmate.
+1. Student chooses to add a tag to a batchmate (could be himself or others).
+2. MassLinkers searches through the list of batchmates for the selected batchmate.
+3. MassLinkers adds tag to the batchmate.
    \
    Use case ends.
 
 **Extensions**
-* 3a. MassLinkers detects an error in the entered command.
-    * 3a1. MassLinkers requests for the correct command.
-    * 3a2. Student enters new command.
+* 2a. The list of batchmates is empty.
+  * a1. MassLinkers warns adding a tag is invalid.
+    \
+    Use case ends.
+* 2b. MassLinkers is unable to find the batchmate.
+    * b1. MassLinkers prompts for new batchmate details. 
+    * Step b1 is repeated until a batchmate is found.
       \
-      Steps 3a1-3a2 are repeated until the data entered are correct.
-      \
-      Use case resumes from step 4.
-
+      Use case resumes from step 2.
 
-* 5a. MassLinkers cannot find classmate in MassLinkers.
-    * 5a1. MassLinkers requests for correct command argument.
-    * 5a2. Student enters new command.
-      \
-      Steps 5a1-5a2 are repeated until the data entered are correct.
-      \
-      Use case resumes from step 5.
-
-
-* a. At any time, Student chooses to close MassLinkers.
-    * a1. MassLinkers updates the written storage file.
-    * a2. MassLinkers closes the program.
+* *a. At any time, Student chooses to close MassLinkers.
+    * a1. MassLinkers updates the storage file.
       \
       Use case ends.
-
-
+    
```

Comment:
I think we also need to add "edit batchmate", "find batchmate", "add module", "edit module".